### ISSUE TITLE
Category query search fails for one particular category in the group.

### ISSUE THREAD

**Issue by whoisjuan** [craftcms/cms#3076]
### Description

This is a very weird issue. I have a template that performs a category query in the following fashion:
{% set platformCategory = craft.categories.group('categoryGroup').search(category).one() %}

I get the category variable from the current URL. I have a total of 4 categories in that group. This query works on all of them except for the last category in the group (I checked this dynamically and by hard-coding the queries). 

After further inspection, I found that the three categories that work have neighboring IDs (14,15,16) and the one that fails is not a neighbor (28). 

If you perform this query {% set platformCategory = craft.categories.group('categoryGroup').id(28).one()  %} Craft will actually return a succesful response that will render the elements based on that category.

It's an extremely weird issue and I have no reasons to suspect this is a bug on my side since I have been able to make queries for 3 of the 4 categories. I already cleared my cache and search indexes with no success.

Please advice.

### Steps to reproduce

See above.

### Additional info

- Craft version: Craft Pro 3.0.14
- PHP version: 7.1.16
- Database driver & version: MySQL 5.7.19
- Plugins & versions:

**Comment by brandonkelly** [comment]
If you search from the category in the Control Panel using  the same  search term as the `category`  variable is getting set to, does the category show up in the search results?

**Comment by whoisjuan** [comment]
Just check that. It doesn't show up.

**Comment by brandonkelly** [comment]
What are you searching for? Can you try re-saving the category?

**Comment by whoisjuan** [comment]
I already did that and it didn't work.  The name of the category and slug is the same. (Name: 'Else', Slug: 'else'). I also tried changing the name of the category and the slug and that didn't work either.

**Comment by brandonkelly** [comment]
Ah guessing this might have to do with it being 4 characters long. By default MySQL’s [ft_min_word_len](https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_ft_min_word_len) setting is set to `4`, but maybe that’s not the case for you, so the term is getting ignored by MySQL.

Try adding this to config/app.php:

```php
return [
    'components' => [
        'search' => [
            'minFullTextWordLength' => 5, // or whatever ft_min_word_len is set to
        ],
    ],
];
```

**Comment by whoisjuan** [comment]
That didn't help. Have in mind that I also have two other categories that are 3 words and they work when I search for them in the CP and when I query them from the template (web and ios). See the GIF below:

![2018-07-08_12-10-54](https://user-images.githubusercontent.com/1447283/42423048-4ac4316e-82a8-11e8-9bc0-34ac4b76d22b.gif)

**Comment by whoisjuan** [comment]
I also just checked the categories table trying to see if there's any corruption in that table, but everything seems fine:
![image](https://user-images.githubusercontent.com/1447283/42423097-4313eed6-82a9-11e8-99e7-ee4ef7c3bf7f.png)

**Comment by whoisjuan** [comment]
Alright. So I did a pretty disgusting fix (I hope that I don't have to stick with this), which makes me think there's something bizarre going on. Here's what I did: 
1) I added a 1 to all the slugs of the categories
![image](https://user-images.githubusercontent.com/1447283/42423280-b29a3546-82ac-11e8-9b37-a01d3f72a519.png)

2) I slice that "1" character when building the variable for the query. I have to do this because an exact match search against that category including "else1" will not work.
`{% set category = craft.app.request.getSegment(2)|slice(0,-1) %} `

A partial string does retrieve the category. So searching "else" against an "else1" slug works.

I don't know if anything of this information would help to find a root issue, but I'm honestly baffled. I don't want to delete that category because that would mean I have to add it again to all the entries that live under it.

Do you have an intuition of what could be the problem? Also is worth noting that I have this issue in both my local dev and production.

**Comment by brandonkelly** [comment]
If you send a database backup and your `composer.json` and `composer.lock` files over to support@craftcms.com, we can try to look into it on our end.

As this doesn’t seem like a normal user-submitted search, have you considered using the `slug` param instead of `search` ? That one will be more reliable, as it never tries to use FULLTEXT searching or make any changes to the user input; it will just be a normal WHERE condition against the `slug` column.

```twig
{% set platformCategory = craft.categories()
    .group('categoryGroup')
    .slug(category)
    .one() %}
```

**Comment by whoisjuan** [comment]
Yes. That's probably a better way to do it. I'll share the files with support soon.
